src/clustering/administration/servers/auto_reconnect.cc

```diff
@@ -84,10 +84,25 @@ void auto_reconnector_t::try_reconnect(const server_id_t &server,
 
     wait_any_t interruptor(&reconnected, &give_up_timer, keepalive.get_drain_signal());
 
+    cond_t join_failed;
     exponential_backoff_t backoff(50, 15 * 1000);
     try {
-        while (!interruptor.is_pulsed()) {
-            connectivity_cluster_run->join(last_known_address, join_delay_secs);
+        while (!interruptor.is_pulsed() && !join_failed.is_pulsed()) {
+            coro_t::spawn_now_dangerously([&]() {
```

Comment:
Note that fixing this might potentially cause dead-locks during shutdown, since `join_blocking` I believe keeps running if the join is successful until the connection is terminated. Depending on how things get shut down, the connections might only get terminated when the `connectivity_cluster_run` is destructed. However `connectivity_cluster_run` can't get destructed as long as the `auto_reconnector_t` is around, and the `auto_reconnector_t` (after the suggested change) cannot be destructed while the coroutine with the `join_blocking` is still running.

I'm not sure what the best solution for this is. If it turns out to be a problem, we can discuss some ideas on Slack maybe.